extmod/machine_i2c_target.c

```diff
@@ -0,0 +1,356 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2025 Damien P. George
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/runtime.h"
+
+#if MICROPY_PY_MACHINE_I2C_TARGET
+
+#include "extmod/modmachine.h"
+#include "shared/runtime/mpirq.h"
+
+enum {
+    I2C_TARGET_IRQ_ADDR_MATCH = 1 << 0,
+    I2C_TARGET_IRQ_READ_REQ = 1 << 1,
+    I2C_TARGET_IRQ_WRITE_REQ = 1 << 2,
+    I2C_TARGET_IRQ_END = 1 << 3,
+    I2C_TARGET_IRQ_STOP = 1 << 4, // should be able to support this on all targets (rp2 needs custom code though), but maybe can't support addr_match?
+
+    I2C_TARGET_IRQ_MEM_ADDR_MATCH = 1 << 5, // needed?
```

Comment:
These look not necessary. The ..START ones are at the same time that the ADDR_MATCH, and the ..END ones are related to STOP or END:

Reply:
Unnecessary events have now been removed.